presto-tests/src/test/java/io/prestosql/execution/EventsAwaitingQueries.java

```diff
@@ -40,19 +42,31 @@
     MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int numEventsExpected, Session session)
             throws Exception
     {
-        return runQueryAndWaitForEvents(sql, numEventsExpected, session, Optional.empty());
+        return runQueryAndWaitForEvents(sql, numEventsExpected, event -> true, session, Optional.empty());
     }
 
     MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int numEventsExpected, Session session, String expectedExceptionRegEx)
             throws Exception
     {
-        return runQueryAndWaitForEvents(sql, numEventsExpected, session, Optional.of(expectedExceptionRegEx));
+        return runQueryAndWaitForEvents(sql, numEventsExpected, event -> true, session, Optional.of(expectedExceptionRegEx));
     }
 
-    private MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int numEventsExpected, Session session, Optional<String> expectedExceptionRegEx)
+    MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int numEventsExpected, Predicate<QueryMetadata> queryFilter, Session session)
             throws Exception
     {
-        eventsCollector.reset(numEventsExpected);
+        return runQueryAndWaitForEvents(sql, numEventsExpected, queryFilter, session, Optional.empty());
+    }
+
+    MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int numEventsExpected, Predicate<QueryMetadata> queryFilter, Session session, String expectedExceptionRegEx)
```

Comment:
sadly even more overloads. Maybe the simplifying one was not a good idea after all?

Reply:
I will drop it.